extensions/omni_txn/omni_txn.c

```diff
@@ -114,6 +114,14 @@ Datum retry(PG_FUNCTION_ARGS) {
     collect_backoff_values = PG_GETARG_BOOL(3);
   }
 
+   // Timeout logic
+  int64 timeout_microsecs = 30000000; // Default timeout value (30 seconds)
+  if (!PG_ARGISNULL(5)) {
+  timeout_microsecs = PG_GETARG_INT64(5); // Retrieve timeout from the argument
```

Comment:
There is `PG_GETARG_INTERVAL_P` – interval is definitely not a 64-bit int but a pointer to [Interval](https://doxygen.postgresql.org/structInterval.html)